### ISSUE TITLE
Programmatic authentication

### ISSUE THREAD

**Issue by adraghici** [gc3-uzh-ch/elasticluster#250]
Hi all,

I am wondering whether there is any way to perform the authentication step programatically (i.e. without needing to follow the URL and entering the verification code manually)?

My problem is that I am trying to find a way to start and manage an Elasticluster GCE fleet from machines where installing Elasticluster itself is not an option.

I am doing this by bootstrapping a GCE instance with Elasticluster installed, which acts as a coordinator node that can take commands to interact with the cluster. However, since I want to automate the creation of the cluster, I need to be able to bypass the manual authentication step and do it in some other way.

I did some reading and looked into the source code and it seems like this is all the basic workflow for OAuth 2.0 for applications accessing Google resources, so it might not be really possible to hack it.

TL;DR: I'm trying to fully automate the deployment of a GCE cluster on behalf of a user who only provides his credentials, but I am currently blocked by the requirement for user interaction imposed by OAuth.

Any ideas or suggestions are much appreciated!

Thanks!

**Comment by riccardomurri** [comment]
Hello Adrian,

I am not an expert with GCE, but they seem to have a notion of "service accounts" for this purpose: https://cloud.google.com/compute/docs/access/create-enable-service-accounts-for-instances

Have you tried asking the same question on the GCE forums?

Ciao,
Riccardo

**Comment by adraghici** [comment]
I read about service accounts, but I was wondering if it was possible to do it somehow wit OAuth since this is the way Elasticluster does authentication. I might try to add support for doing this using service accounts over the following days, so I'll keep this issue open for now.

Thanks!

**Comment by riccardomurri** [comment]
I think this is now fixed in PR #616 by @ndmckinley  -- please re-open if you think that doesn't cover your use-case.
